dist/jsonapi-client-test.js

```diff
@@ -16567,45 +16568,46 @@ var client = new Client("http://localhost:16006/rest");
 
 describe("Testing jsonapi-client", function() {
 
-  it("creates a resource", function(done) {
-    var newPerson = client.create("people");
-    newPerson.set("email", "foo@bar.com");
+  context("supports promises", function() {
 
-    assert.deepEqual(newPerson.toJSON(), {
-      id: null,
-      type: "people",
-      email: "foo@bar.com"
+    it("client.find", function(done) {
+      client.find("people", { }).then(function(people) {
+        assert.ok(people[0] instanceof Client.Resource);
+      }).then(done, done);
     });
 
-    newPerson.sync(function(err1) {
-      assert.equal(err1, null);
-      assert.ok(newPerson._getUidString().match(/[a-z0-9-]+\/people/));
+    it("client.get", function(done) {
+      client.get("people", "d850ea75-4427-4f81-8595-039990aeede5", { }).then(function(person) {
+        assert.ok(person instanceof Client.Resource);
+      }).then(done, done);
+    });
 
-      client.get("people", newPerson._getUid(), { }, function(err2, somePerson) {
-        assert.equal(err2, null);
-        assert.equal(newPerson, somePerson);
+    it("client.fetch", function(done) {
+      client.get("people", "d850ea75-4427-4f81-8595-039990aeede5", { }).then(function(person) {
+        return person.fetch("articles");
+      }).then(function(articles) {
+        assert.ok(articles[0] instanceof Client.Resource);
+      }).then(done, done);
+    });
 
+    it("create-sync-delete", function(done) {
+      var newPerson = client.create("people");
+      var uuid;
+      newPerson.sync().then(function() {
+        uuid = newPerson._getUid();
+        return newPerson.delete();
+      }).then(function() {
+        return client.get("people", uuid, { });
+      }).then(function() {
+        throw new Error("Should have errored!");
+      }, function(err) {
+        assert.ok(err.message.match(/There is no people with id /));
         done();
       });
```

Comment:
The `.then(success_callback, error_callback)` is fine for the `.then(done, done)` cases because it's more compact, here however I would find it more readable if we used the `.then(success_callback).catch(error_callback)` form instead.